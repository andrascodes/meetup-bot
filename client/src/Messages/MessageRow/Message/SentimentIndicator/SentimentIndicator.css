.Positive {
  color: green;
}

.Icon {
  margin-left: 0.2em;
}

.Negative {
  color: red;
}