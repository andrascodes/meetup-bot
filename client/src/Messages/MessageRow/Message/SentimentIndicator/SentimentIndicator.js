import React from 'react'

import './SentimentIndicator.css'

export const SentimentIndicator = props => {
  if(!props.value) {
    return (<div></div>)
  }
  
  const valueMap = {
    positive: {
      className: 'Positive',
      text: 'positive',
      icon: 'fa-check',
      style: {
        
      }
    },
    neutral: {
      className: 'Neutral',
      text: 'neutral',
      icon: 'fa-check'
    },
    negative: {
      className: 'Negative',
      text: 'negative',
      icon: 'fa-times'
    }
  }

  const { text, icon, className } = valueMap[props.value]

  return (
    <div className="Sentiment">
      {`Sentiment: `}
      <span className={`${className}`}>{text}</span>
      <i className={`fa ${icon} ${className} Icon`}></i>
    </div>
  )
}