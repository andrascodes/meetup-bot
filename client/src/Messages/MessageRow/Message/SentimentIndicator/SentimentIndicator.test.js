import React from 'react'
import renderer from 'react-test-renderer'

import { SentimentIndicator } from './SentimentIndicator'

describe('SentimentIndicator', () => {

  test('renders correctly', () => {
    let tree = renderer.create(
      <SentimentIndicator
        value={'positive'}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
    
    tree = renderer.create(
      <SentimentIndicator
        value={'negative'}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
    
    tree = renderer.create(
      <SentimentIndicator
        value={'neutral'}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders correctly when value is falsy', () => {
    const tree = renderer.create(
      <SentimentIndicator 
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

})