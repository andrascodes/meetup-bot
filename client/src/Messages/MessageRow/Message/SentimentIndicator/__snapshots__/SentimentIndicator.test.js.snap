exports[`SentimentIndicator renders correctly 1`] = `
<div
  className="Sentiment">
  Sentiment: 
  <span
    className="Positive">
    positive
  </span>
  <i
    className="fa fa-check Positive Icon" />
</div>
`;

exports[`SentimentIndicator renders correctly 2`] = `
<div
  className="Sentiment">
  Sentiment: 
  <span
    className="Negative">
    negative
  </span>
  <i
    className="fa fa-times Negative Icon" />
</div>
`;

exports[`SentimentIndicator renders correctly 3`] = `
<div
  className="Sentiment">
  Sentiment: 
  <span
    className="Neutral">
    neutral
  </span>
  <i
    className="fa fa-check Neutral Icon" />
</div>
`;

exports[`SentimentIndicator renders correctly when value is falsy 1`] = `<div />`;
